src/reader_struct.h

```diff
@@ -181,7 +181,7 @@ struct Primitive<std::vector<T>> {
 		typename std::vector<T>::iterator it;
 		printf("  ");
 		for (it = ref.begin(); it != ref.end(); ++it) {
-			printf("%d, ", *it);
+			printf("%d, ", (int)*it);
```

Comment:
I like C-style casts, however, as liblcf is C++, there should be a fancy, type-aware c++ variant of a cast, maybe.